src/main/scala/com/sky/kafka/topicloader/config/config.scala

```diff
@@ -16,7 +17,8 @@ package object config {
   implicit class EitherOps[A](e: Either[Throwable, A]) {
```

Comment:
this is only called in `Config.load` for `PosInt`s. So I would either change this syntax extension so it takes an `Either[IllegalArgumentException, A]` or I'd change the `PosInt` apply to return `Either[ConfigException.BadValue, PosInt]`

Then you don't have to worry about catching `Throwable` anywhere.

If you go with the option of changing the syntax extension method, I think it should also be made package private (I dunno if it was private before though, or if it even existed before)

Reply:
Went with option 1, made it private. Still have to catch a nonFatal in the TryOps as the Try will always throw a ConfigException but it's unboxed so have to re-wrap it